pkg/operator/controller/controller_ca_configmap.go

```diff
@@ -0,0 +1,163 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	ingressv1alpha1 "github.com/openshift/cluster-ingress-operator/pkg/apis/ingress/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	// GlobalMachineSpecifiedConfigNamespace is the location for global
+	// config.  In particular, the operator will put the configmap with the
+	// CA certificate in this namespace.
+	GlobalMachineSpecifiedConfigNamespace = "openshift-config-managed"
+
+	// caCertConfigMapName is the name of the config map with the public key
+	// for the CA certificate, which the operator publishes for other
+	// operators to use.
+	caCertConfigMapName = "router-ca"
+)
+
+// routerCAConfigMapName returns the namespaced name for the router CA
+// configmap.
+func routerCAConfigMapName() types.NamespacedName {
+	return types.NamespacedName{
+		Namespace: GlobalMachineSpecifiedConfigNamespace,
+		Name:      caCertConfigMapName,
+	}
+}
+
+// ensureRouterCAConfigMap will create, update, or delete the configmap for the
+// router CA as appropriate.
+func (r *reconciler) ensureRouterCAConfigMap(ingresses []ingressv1alpha1.ClusterIngress) error {
+	desired, err := r.desiredRouterCAConfigMap(ingresses)
+	if err != nil {
+		return err
+	}
+	current, err := r.currentRouterCAConfigMap()
+	if err != nil {
+		return err
+	}
+	switch {
+	case desired == nil && current == nil:
+		// Nothing to do.
+	case desired == nil && current != nil:
+		if err := r.deleteRouterCAConfigMap(current); err != nil {
+			return fmt.Errorf("failed to ensure router CA was unpublished: %v", err)
+		}
+	case desired != nil && current == nil:
+		if err := r.createRouterCAConfigMap(desired); err != nil {
+			return fmt.Errorf("failed to ensure router CA was published: %v", err)
+		}
+	case desired != nil && current != nil:
+		if err := r.updateRouterCAConfigMap(current, desired); err != nil {
+			return fmt.Errorf("failed to update published router CA: %v", err)
+		}
+	}
+	return nil
+}
+
+// desiredRouterCAConfigMap returns the desired router CA configmap.
+func (r *reconciler) desiredRouterCAConfigMap(ingresses []ingressv1alpha1.ClusterIngress) (*corev1.ConfigMap, error) {
+	if !shouldPublishRouterCA(ingresses) {
+		return nil, nil
+	}
+
+	secret, err := r.ensureRouterCACertificateSecret()
```

Comment:
The pattern from the other refactors is to explicitly accept dependencies through the function arguments.

Reply:
Done.  Thanks!